PWGCF/TwoParticleCorrelations/Tasks/lambdaR2Correlation.cxx

```diff
@@ -356,49 +465,26 @@ struct lambdaCorrTableProducer {
         continue;
       }
 
-      auto postrack = v0.template posTrack_as<Tracks>();
-      auto negtrack = v0.template negTrack_as<Tracks>();
-
-      // get lambda
-      if ((fabs(v0.mLambda() - MassLambda0) < cfg_lambda_mass_window) && (fabs(v0.mK0Short() - MassK0Short) > cfg_kshort_rej) && (selPIDTrack(postrack, postrack.tpcNSigmaPr(), postrack.tofNSigmaPr())) && (selPIDTrack(negtrack, negtrack.tpcNSigmaPi(), negtrack.tofNSigmaPi()))) {
-        fillQALambda<kLambda>(collision, v0, tracks);
-        lambdaTrackTable(lambdaCollisionTable.lastIndex(), v0.pt(), v0.yLambda(), v0.eta(), v0.phi(), v0.mLambda(), postrack.index(), negtrack.index(), true);
-      }
-
-      // get anti-lambda
-      if ((fabs(v0.mAntiLambda() - MassLambda0) < cfg_lambda_mass_window) && (fabs(v0.mK0Short() - MassK0Short) > cfg_kshort_rej) && (selPIDTrack(negtrack, negtrack.tpcNSigmaPr(), negtrack.tofNSigmaPr())) && (selPIDTrack(postrack, postrack.tpcNSigmaPi(), postrack.tofNSigmaPi()))) {
-        fillQALambda<kAntiLambda>(collision, v0, tracks);
-        lambdaTrackTable(lambdaCollisionTable.lastIndex(), v0.pt(), v0.yLambda(), v0.eta(), v0.phi(), v0.mAntiLambda(), postrack.index(), negtrack.index(), false);
-      }
+      selV0Particle<kLambda, kProton, kPion, kCentralWindow>(collision, v0, tracks);
+      selV0Particle<kLambda, kProton, kPion, kLeftWindow>(collision, v0, tracks);
+      selV0Particle<kLambda, kProton, kPion, kRightWindow>(collision, v0, tracks);
+      selV0Particle<kAntiLambda, kPion, kProton, kCentralWindow>(collision, v0, tracks);
+      selV0Particle<kAntiLambda, kPion, kProton, kLeftWindow>(collision, v0, tracks);
+      selV0Particle<kAntiLambda, kPion, kProton, kRightWindow>(collision, v0, tracks);
```

Comment:
Why it is needed to do the six combinations if only one will be valid?
Check my comments on the routine